worker/include/Channel/Request.hpp

```diff
@@ -76,15 +76,18 @@ namespace Channel
 		void TypeError(const char* reason = nullptr);
 
 	public:
-		// Passed by argument.
-		Channel::UnixStreamSocket* channel{ nullptr };
 		uint32_t id{ 0u };
 		std::string method;
 		MethodId methodId;
 		json internal;
 		json data;
 		// Others.
 		bool replied{ false };
+
+	public:
+		static void ClassInit(Channel::UnixStreamSocket* channel);
+		// Passed by argument.
+		static Channel::UnixStreamSocket* channel;
```

Comment:
Why has this been converted into a `static` member? Also, it's no longer "passed by argument", right?

If so, it must be separated from the static methods and added into a new `public:` group below just for variable static members. However, first I need to understand why it's now a `static` member.

Reply:
now that we use two separate pipe. After getting request from **Consumer Channel**,  we'll send response ( accept or reject) to JS which should use **Producer Channel**, but **Consumer Channel* has not relation to **Producer Channel**, so use a `static` member store the  **Producer Channel** just like` Channel::Notifier`.

Reply:
I understand, however I'd prefer to avoid static members as far as possible. I've seen that you have created a `ChannelWrapper` class which holds both the Producer Channel and Consumer Channel. Perhaps we could manage a single instance of `ChannelWrapper` and pass it to other classes.

Reply:
The problem is that `ChannelWrapper` depends on `Producer Channel ` and `Consumer Channel`, `Consumer Channel` create `request` object. so making a single instance of `ChannelWrapper` and passing it `Consumer Channel` seems weird.

If we wan't to avoid static members, there are two approaches I can think of.
* wrap `consumerFd` and `producerFd` into `::UnixStreamSocket`, when invoking `read` or `write` we chose the right one.  I already implemented  this [here](https://github.com/haiyangwu/mediasoup-win/commit/5892905f8f22cf479848f5823aa340c0ab0692ee)
* create two subclass (`::ConsumerSocket` and `::ProducerSocket` ) of `::UnixStreamSocket`. `Channel::UnixStreamSocket` instead  of inheriting `::UnixStreamSocket`, it composite `::ConsumerSocket` and `::ProducerSocket`

IMO, I like the second approach.  I need some suggestion from you. 😘

Reply:
I think you were right and we can have a static `channel` member in `Request.hpp` (as we have in `Notifier.hpp`.  Just please ensure both are written in a similar way.

Reply:
>* create two subclass (`::ConsumerSocket` and `::ProducerSocket` ) of `::UnixStreamSocket`. `Channel::UnixStreamSocket` instead  of inheriting `::UnixStreamSocket`, it composite `::ConsumerSocket` and `::ProducerSocket`

[Implementation](https://github.com/haiyangwu/mediasoup-win/commit/6fbcc2aff91dc42ad8044935fe308a8d21a3990a). IMO, this seems better.

Reply:
I've done some comments in that commit: https://github.com/haiyangwu/mediasoup-win/commit/6fbcc2aff91dc42ad8044935fe308a8d21a3990a

Reply:
Specially [this one](https://github.com/haiyangwu/mediasoup-win/commit/6fbcc2aff91dc42ad8044935fe308a8d21a3990a#r35040725).

BTW I like your new approach. I've just "cleaned" it a bit to follow mediasoup guidelines :)